js/src/rive.ts

```diff
@@ -0,0 +1,965 @@
+  
+  // Singleton helpers
+  private static runtime: typeof Runtime;
+  // Flag to indicate that loading has started/completed
+  private static isLoading: boolean = false;
+  // List of callbacks for the runtime that come in while loading
+  private static callBackQueue: RuntimeCallback[] = [];
+  // Instance of the Rive runtime
+  private static rive: typeof Runtime;
+  // The url for the Wasm file
+  private static wasmWebPath: string = 'https://unpkg.com/rive-js@latest/dist/';
+  // Local path to the Wasm file; for testing purposes
+  private static wasmFilePath: string = 'dist/';
+  // Are we in test mode?
+  private static testMode: boolean = false;
+
+  // Class is never instantiated
+  private constructor() {}
+
+  // Loads the runtime
+  private static loadRuntime() : void {
+    Runtime({
+      // Loads Wasm bundle
+      locateFile: (file: string) =>
+        // if in test mode, attempts to load file locally 
+        (RuntimeLoader.testMode ?
+          RuntimeLoader.wasmFilePath :
+          RuntimeLoader.wasmWebPath) + file
+    }).then((rive: typeof Runtime) => {
+      RuntimeLoader.runtime = rive;
+      // Fire all the callbacks
+      while (RuntimeLoader.callBackQueue.length > 0) {
+        RuntimeLoader.callBackQueue.shift()(RuntimeLoader.runtime);
+      }
+    });
+  }
+
+  // Provides a runtime instance via a callback
+  public static getInstance(callback: RuntimeCallback): void {
+    // If it's not loading, start loading runtime
+    if (!RuntimeLoader.isLoading) {
+      RuntimeLoader.isLoading = true;
+      RuntimeLoader.loadRuntime();
+    }
+    if (!RuntimeLoader.runtime) {
+      RuntimeLoader.callBackQueue.push(callback);
+    } else {
+      callback(RuntimeLoader.runtime);
+    }
+  }
+
+  // Provides a runtime instance via a promise
+  public static awaitInstance(): Promise<typeof Runtime> {
+    return new Promise<typeof Runtime>((resolve, reject) => 
+      RuntimeLoader.getInstance((rive: typeof Runtime): void => resolve(rive))
+    );
+  }
+
+  // Places the loader in test mode
+  public static setTestMode(mode: boolean): void {
+    RuntimeLoader.testMode = mode;
+  }
+}
+
+// #endregion
+
+// #region animations
+
+// Wraps animations and instances from the runtime and keeps track of playback
+// state
+export class Animation {  
+  public loopCount: number = 0;
+  public paused: boolean = false;
+
+  constructor(private animation: any, public readonly instance: any) {}
+
+  // Returns the animation's name
+  public get name(): string {
+    return this.animation.name;
+  }
+
+  // Returns the animation's loop type
+  public get loopValue(): number {
+    return this.animation.loopValue;
+  } 
+}
+
+// #endregion
+
+// #region events
+
+// Events that Rive fires
+export enum EventType {
+  Load      = 'load',
+  LoadError = 'loaderror',
+  Play      = 'play',
+  Pause     = 'pause',
+  Stop      = 'stop',
+  Loop      = 'loop',
+}
+
+// Event fired by Rive
+export interface Event {
```

Comment:
`Event` is already a type of the DOM API, overriding DOM API types might lead to unexpected side effect (wrong errors in the compiler). If it's exposed I would recommend more specific name like `RiveEvent`.

Reply:
Good point; I'll see about a better name.